lib/graphql/execution/lazy/resolve.rb

```diff
@@ -35,12 +35,17 @@ def self.resolve_in_place(value)
           if acc.empty?
             Lazy::NullResult
           else
-            Lazy.new {
-              acc.each_with_index { |ctx, idx|
-                acc[idx] = ctx.value.value
-              }
-              resolve_in_place(acc)
-            }
+            Lazy.new(
+              value: -> {
+                acc.each { |ctx| ctx.value.execute }
```

Comment:
This is the important part. In the resolution, `execute` is called on each lazy object before the values are resolved. This allows concurrent execution to begin in parallel for multiple objects in the same graph layer.